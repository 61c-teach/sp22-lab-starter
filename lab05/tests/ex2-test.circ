<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project source="3.4.3-61c.cc0f4a6" version="1.0">
  This file is intended to be loaded by Logisim-evolution (https://github.com/reds-heig/logisim-evolution).

  <lib desc="#Wiring" name="0">
    <tool name="Splitter">
      <a name="appear" val="center"/>
      <a name="facing" val="west"/>
    </tool>
    <tool name="Pin">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Probe">
      <a name="radix" val="16"/>
    </tool>
    <tool name="Tunnel">
      <a name="width" val="2"/>
    </tool>
    <tool name="Pull Resistor">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Clock">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Constant">
      <a name="value" val="0xff"/>
      <a name="width" val="8"/>
    </tool>
  </lib>
  <lib desc="#Gates" name="1">
    <tool name="PLA">
      <a name="table" val=""/>
    </tool>
  </lib>
  <lib desc="#Plexers" name="2">
    <tool name="Multiplexer">
      <a name="enable" val="false"/>
    </tool>
    <tool name="Demultiplexer">
      <a name="enable" val="false"/>
    </tool>
  </lib>
  <lib desc="#Arithmetic" name="3"/>
  <lib desc="#Memory" name="4">
    <tool name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="T Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="J-K Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="S-R Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Counter">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Shift Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Random">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="RAM">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="ROM">
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 8 8
0
</a>
    </tool>
  </lib>
  <lib desc="#Base" name="5">
    <tool name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
  </lib>
  <lib desc="file#../ex2.circ" name="6"/>
  <main name="main"/>
  <options>
    <a name="gateUndefined" val="ignore"/>
    <a name="simlimit" val="1000"/>
    <a name="simrand" val="0"/>
  </options>
  <mappings>
    <tool lib="5" map="Button2" name="Menu Tool"/>
    <tool lib="5" map="Button3" name="Menu Tool"/>
    <tool lib="5" map="Ctrl Button1" name="Menu Tool"/>
  </mappings>
  <toolbar>
    <tool lib="5" name="Poke Tool"/>
    <tool lib="5" name="Edit Tool"/>
    <tool lib="5" name="Wiring Tool"/>
    <tool lib="5" name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
    <sep/>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
    </tool>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
    </tool>
    <sep/>
    <tool lib="1" name="NOT Gate"/>
    <tool lib="1" name="AND Gate"/>
    <tool lib="1" name="OR Gate"/>
    <tool lib="1" name="XOR Gate"/>
    <tool lib="1" name="NAND Gate"/>
    <tool lib="1" name="NOR Gate"/>
    <sep/>
    <tool lib="4" name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool lib="4" name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
  </toolbar>
  <circuit name="main">
    <a name="appearance" val="logisim_evolution"/>
    <a name="circuit" val="main"/>
    <a name="circuitnamedboxfixedsize" val="true"/>
    <a name="circuitvhdlpath" val=""/>
    <a name="clabel" val=""/>
    <a name="clabelfont" val="SansSerif bold 16"/>
    <a name="clabelup" val="east"/>
    <comp lib="0" loc="(110,140)" name="Clock"/>
    <comp lib="0" loc="(190,100)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="CYCLE"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
      <a name="width" val="3"/>
    </comp>
    <comp lib="0" loc="(190,50)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="halt"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(280,240)" name="Splitter">
      <a name="appear" val="right"/>
      <a name="fanout" val="6"/>
      <a name="incoming" val="6"/>
    </comp>
    <comp lib="0" loc="(480,200)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="A"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(480,230)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="B"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(480,260)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="C"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(480,290)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="D"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(480,320)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="SEL0"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(480,350)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="SEL1"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(60,110)" name="Constant">
      <a name="width" val="3"/>
    </comp>
    <comp lib="0" loc="(60,40)" name="Constant">
      <a name="value" val="0x7"/>
      <a name="width" val="3"/>
    </comp>
    <comp lib="0" loc="(720,390)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="NAND_OUT"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(720,460)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="NOR_OUT"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(720,530)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="XOR_OUT"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(720,600)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="MUX2_OUT"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(720,690)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="MUX4_OUT"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="3" loc="(100,100)" name="Adder">
      <a name="width" val="3"/>
    </comp>
    <comp lib="3" loc="(100,50)" name="Comparator">
      <a name="width" val="3"/>
    </comp>
    <comp lib="4" loc="(110,70)" name="Register">
      <a name="appearance" val="logisim_evolution"/>
      <a name="width" val="3"/>
    </comp>
    <comp lib="4" loc="(30,180)" name="ROM">
      <a name="addrWidth" val="3"/>
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 3 6
24 12 1 37 0 3f 15 2a
</a>
      <a name="dataWidth" val="6"/>
    </comp>
    <comp lib="6" loc="(700,390)" name="NAND2"/>
    <comp lib="6" loc="(700,460)" name="NOR2"/>
    <comp lib="6" loc="(700,530)" name="XOR2"/>
    <comp lib="6" loc="(700,600)" name="MUX2"/>
    <comp lib="6" loc="(700,690)" name="MUX4"/>
    <wire from="(100,100)" to="(110,100)"/>
    <wire from="(100,50)" to="(190,50)"/>
    <wire from="(170,100)" to="(180,100)"/>
    <wire from="(180,100)" to="(180,170)"/>
    <wire from="(180,100)" to="(190,100)"/>
    <wire from="(270,240)" to="(280,240)"/>
    <wire from="(30,170)" to="(180,170)"/>
    <wire from="(30,170)" to="(30,190)"/>
    <wire from="(30,60)" to="(30,90)"/>
    <wire from="(30,60)" to="(60,60)"/>
    <wire from="(30,90)" to="(30,170)"/>
    <wire from="(30,90)" to="(60,90)"/>
    <wire from="(300,250)" to="(320,250)"/>
    <wire from="(300,260)" to="(330,260)"/>
    <wire from="(300,270)" to="(340,270)"/>
    <wire from="(300,280)" to="(340,280)"/>
    <wire from="(300,290)" to="(330,290)"/>
    <wire from="(300,300)" to="(320,300)"/>
    <wire from="(320,200)" to="(320,250)"/>
    <wire from="(320,200)" to="(360,200)"/>
    <wire from="(320,300)" to="(320,350)"/>
    <wire from="(320,350)" to="(460,350)"/>
    <wire from="(330,230)" to="(330,260)"/>
    <wire from="(330,230)" to="(380,230)"/>
    <wire from="(330,290)" to="(330,320)"/>
    <wire from="(330,320)" to="(440,320)"/>
    <wire from="(340,260)" to="(340,270)"/>
    <wire from="(340,260)" to="(400,260)"/>
    <wire from="(340,280)" to="(340,290)"/>
    <wire from="(340,290)" to="(420,290)"/>
    <wire from="(360,200)" to="(360,390)"/>
    <wire from="(360,200)" to="(480,200)"/>
    <wire from="(360,390)" to="(360,460)"/>
    <wire from="(360,390)" to="(480,390)"/>
    <wire from="(360,460)" to="(360,530)"/>
    <wire from="(360,460)" to="(480,460)"/>
    <wire from="(360,530)" to="(360,600)"/>
    <wire from="(360,530)" to="(480,530)"/>
    <wire from="(360,600)" to="(360,690)"/>
    <wire from="(360,600)" to="(480,600)"/>
    <wire from="(360,690)" to="(480,690)"/>
    <wire from="(380,230)" to="(380,410)"/>
    <wire from="(380,230)" to="(480,230)"/>
    <wire from="(380,410)" to="(380,480)"/>
    <wire from="(380,410)" to="(480,410)"/>
    <wire from="(380,480)" to="(380,550)"/>
    <wire from="(380,480)" to="(480,480)"/>
    <wire from="(380,550)" to="(380,620)"/>
    <wire from="(380,550)" to="(480,550)"/>
    <wire from="(380,620)" to="(380,710)"/>
    <wire from="(380,620)" to="(480,620)"/>
    <wire from="(380,710)" to="(480,710)"/>
    <wire from="(400,260)" to="(400,730)"/>
    <wire from="(400,260)" to="(480,260)"/>
    <wire from="(400,730)" to="(480,730)"/>
    <wire from="(420,290)" to="(420,750)"/>
    <wire from="(420,290)" to="(480,290)"/>
    <wire from="(420,750)" to="(480,750)"/>
    <wire from="(440,320)" to="(440,640)"/>
    <wire from="(440,320)" to="(480,320)"/>
    <wire from="(440,640)" to="(440,770)"/>
    <wire from="(440,640)" to="(480,640)"/>
    <wire from="(440,770)" to="(480,770)"/>
    <wire from="(460,350)" to="(460,790)"/>
    <wire from="(460,350)" to="(480,350)"/>
    <wire from="(460,790)" to="(480,790)"/>
    <wire from="(700,390)" to="(720,390)"/>
    <wire from="(700,460)" to="(720,460)"/>
    <wire from="(700,530)" to="(720,530)"/>
    <wire from="(700,600)" to="(720,600)"/>
    <wire from="(700,690)" to="(720,690)"/>
  </circuit>
</project>
