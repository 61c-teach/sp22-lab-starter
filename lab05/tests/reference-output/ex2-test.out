CYCLE	A	B	C	D	SEL0	SEL1	NAND_OUT	NOR_OUT	XOR_OUT	MUX2_OUT	MUX4_OUT
000	0	0	1	0	0	1	1	1	0	0	1
001	0	1	0	0	1	0	1	0	1	1	1
010	1	0	0	0	0	0	1	0	1	1	1
011	1	1	1	0	1	1	0	0	0	1	0
100	0	0	0	0	0	0	1	1	0	0	0
101	1	1	1	1	1	1	0	0	0	1	1
110	1	0	1	0	1	0	1	0	1	0	0
111	0	1	0	1	0	1	1	0	1	0	0
